solana/pyth2wormhole/program/src/attest.rs

```diff
@@ -248,38 +262,118 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
-            AccountMeta::new_readonly(*accs.clock.info().key, false),
-            AccountMeta::new_readonly(solana_program::sysvar::rent::ID, false),
-            AccountMeta::new_readonly(solana_program::system_program::id(), false),
-        ],
-    );
-
-    trace!("Before cross-call");
+    let payload = batch_attestation.serialize().map_err(|e| {
+        trace!(&e.to_string());
+        ProgramError::InvalidAccountData
+    })?;
+
+    // Adjust message account size if necessary
+    let ix = if accs.wh_message.is_initialized() {
+        if accs.wh_message.payload.len() != payload.len() {
+            // Learn how much we're adjusting (Note: assuming surrounding
+            // Wormhole message type is constant-size).
+            // positive -> grow,
+            // negative -> shrink,
+            let diff = payload.len() as i64 - accs.wh_message.payload.len() as i64;
+            accs.wh_message.0 .0.realloc(
+                (accs.wh_message.payload.len() as i64 + diff) as usize,
+                false,
+            )?;
+        }
+        trace!("After message size adjustment");
+
+        // Send payload
+        let post_message_unreliable_data = (
+            bridge::instruction::Instruction::PostMessageUnreliable,
+            PostMessageData {
+                nonce: 0, // Superseded by the sequence number
+                payload,
+                consistency_level: data.consistency_level,
+            },
+        );
+
+        trace!("Before unreliable cross-call");
+
+        Instruction::new_with_bytes(
+            accs.config.wh_prog,
+            post_message_unreliable_data.try_to_vec()?.as_slice(),
+            vec![
+                AccountMeta::new(*accs.wh_bridge.key, false),
+                AccountMeta::new(*(accs.wh_message.0).0.key, true),
+                AccountMeta::new_readonly(*accs.wh_emitter.key, true),
+                AccountMeta::new(*accs.wh_sequence.key, false),
+                AccountMeta::new(*accs.payer.key, true),
+                AccountMeta::new(*accs.wh_fee_collector.key, false),
+                AccountMeta::new_readonly(*accs.clock.info().key, false),
+                AccountMeta::new_readonly(solana_program::sysvar::rent::ID, false),
+                AccountMeta::new_readonly(solana_program::system_program::id(), false),
+            ],
+        )
+    } else {
+        let post_message_data = (
+            bridge::instruction::Instruction::PostMessage,
+            PostMessageData {
+                nonce: 0,
+                payload,
+                consistency_level: data.consistency_level,
+            },
+        );
+
+        trace!("Before reliable cross-call");
+        Instruction::new_with_bytes(
+            accs.config.wh_prog,
+            post_message_data.try_to_vec()?.as_slice(),
+            vec![
+                AccountMeta::new(*accs.wh_bridge.key, false),
+                AccountMeta::new(*(accs.wh_message.0).0.key, true),
```

Comment:
```suggestion
                AccountMeta::new(*accs.wh_message.info().key, true),
```

Reply:
removed in favor of the upstream instruction generator